client_update.go

```diff
@@ -86,8 +86,10 @@ func (u *UpdateClient) FetchUpdate(api ApiRequester, url string) (io.ReadCloser,
 	}
 
 	if r.ContentLength < 0 {
+		defer r.Body.Close()
```

Comment:
Why we can not have just one `defer r.Body.Close()` after (or before) `log.Debugf("Received fetch update response %v+", r)`?